packages/webdriverio/src/commands/browser/getPuppeteer.ts

```diff
@@ -48,10 +48,21 @@ export default async function getPuppeteer (this: WebdriverIO.Browser) {
         return this.puppeteer
     }
 
+    const caps = (this.capabilities as Capabilities.W3CCapabilities).alwaysMatch || this.capabilities as Capabilities.DesiredCapabilities
+    /**
+     * attach to a Selenium 4 CDP Session if it's returned in the capabilities
+     */
+    const cdpEndpoint = caps['se:options']?.cdp
+    if (cdpEndpoint) {
+        this.puppeteer = await puppeteer.connect({
+            browserWSEndpoint: cdpEndpoint
```

Comment:
If we make some hardcoded assumptions here, this could work with the docker images as described in https://github.com/webdriverio/webdriverio/issues/6470#issuecomment-786109079

Something like:
`ws://${this.config.hostname}:${this.config.port}/session/${this.sessionId}/se/cdp`

Definitely not as slick though and more likely to break in a future update. Would be a temp bandaid until https://github.com/SeleniumHQ/selenium/issues/9202 is addressed.